GVRf/Framework/backend_monoscopic/src/main/java/org/gearvrf/MonoscopicViewManager.java

```diff
@@ -215,7 +215,7 @@ public void surfaceCreated(SurfaceHolder surfaceHolder) {
                         mInputManager.setScene(scene);
                         mRotationSensor.onResume();
                         long vulkanCoreObj;
-                        vulkanCoreObj = NativeVulkanCore.getInstance(vulkanSurfaceView.getHolder().getSurface());
+                        vulkanCoreObj = NativeVulkanCore.getInstance(vulkanSurfaceView.getHolder().getSurface(), true);
```

Comment:
Instead of this surfaceInit flag why not directly call NativeVulkanCore_recreateSwapchain from vulkanSurfaceView...surfaceCreated? Seems to me like it would clearer and simpler.

Reply:
yes, in java we could check if a vulkan instance exists before the getInstance call and call recreateswapchain if the instance exists. I did this in the c++ side right now.